torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -1484,6 +1484,129 @@ TensorView* gather(
   return out_tv;
 }
 
+namespace {
+
+//! Create new output for mma
+static TensorView* newForMma(
+    TensorView* tv_a,
+    TensorView* tv_b,
+    const std::vector<unsigned int>& axes,
+    DataType data_type = DataType::Float) {
+  auto orig_domain_a =
+      TensorDomain::noReductions(tv_a->getMaybeRFactorDomain());
+  auto orig_domain_b =
+      TensorDomain::noReductions(tv_b->getMaybeRFactorDomain());
```

Comment:
Thanks for pointing this out. There is an interface decision involved here since the broadcast cannot be as implicit as pointwise ops:

The position of broadcasts actually implies the input layout, i.e TT/TN/NT, and so we'd need user to either annotate the broadcast or specify a layout so that we can figure out which axis is broadcast.

Reply:
I don't follow this, maybe we need to chat about this more.

Reply:
Addressing in follow up. resolving.